src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py

```diff
@@ -51,13 +52,30 @@ class MCP2221:
     def __init__(self):
         self._hid = hid.device()
         self._hid.open(MCP2221.VID, MCP2221.PID)
+        # make sure the device gets closed before exit
+        atexit.register(self.close)
         if MCP2221_RESET_DELAY >= 0:
-            self._reset()
+            try:
+                self._reset()
+            except OSError:
+                # this might fail on Linux with the hid_mcp2221 native
+                # driver and a short reset delay -- if it fails,
+                # close the device before reraising
+                self.close()
+                raise
```

Comment:
Please revert this part back to original reset code since it seems specific to the `hid_mcp2221` module. The reset delay can be increased via an environment variable `BLINKA_MCP2221_RESET_DELAY`.

Reply:
I just tested again with the `hid_mcp2221` driver and the comment is indeed a bit misleading.  I checked that:
* sending the reset signal just after opening works without errors, and closes the device, so the `close()` after this point is not necessary:
  ```py
  >>> import hid; h = hid.device()
  >>> h.open(0x04D8, 0x00DD); report = b"\x70\xAB\xCD\xEF"; h.write(b"\0" + report + b"\0" * (64 - len(report)))
  65
  >>>
  ```
* trying to open the device again just after the reset signal fails with an `OSError` if there isn't enough delay, however the device is still considered closed, and if the `open()` is not successful the call to `.close()` is not necessary.

This `try`/`except` would still be useful is in case sending the reset signal fails, raising an error and leaving the device open, but I haven't encountered this situation.  Calling `.close()` on a closed device is a no-op though, so I initially added it just as an extra precaution.

Let me know if you still want me to revert the change or just update the comment to explain that the `self.close()` call is just there to guarantee that the device is indeed closed before re-raising.